protocols/imap/src/test/java/org/apache/james/imap/processor/base/SelectedMailboxImplTest.java

```diff
@@ -0,0 +1,172 @@
+    private MessageManager messageManager;
+    private MailboxPath mailboxPath;
+    private ImapSession imapSession;
+    private Mailbox mailbox;
+
+    @Before
+    public void setUp() throws Exception {
+        executorService = Executors.newFixedThreadPool(1);
+
+        mailboxPath = new MailboxPath(MailboxConstants.USER_NAMESPACE, "tellier@linagora.com", MailboxConstants.INBOX);
+        mailboxManager = mock(MailboxManager.class);
+        messageManager = mock(MessageManager.class);
+        when(mailboxManager.getMailbox(eq(mailboxPath), any(MailboxSession.class))).thenReturn(messageManager);
+        when(messageManager.getApplicableFlags(any(MailboxSession.class))).thenReturn(new Flags());
+        when(messageManager.search(any(SearchQuery.class), any(MailboxSession.class)))
+            .then(sleepThenSearchAnswer());
+
+        imapSession = mock(ImapSession.class);
+        when(imapSession.getAttribute(ImapSessionUtils.MAILBOX_SESSION_ATTRIBUTE_SESSION_KEY)).thenReturn(mock(MailboxSession.class));
+
+        mailbox = mock(Mailbox.class);
+        when(mailbox.generateAssociatedPath()).thenReturn(mailboxPath);
+    }
+
+    @After
+    public void tearDown() {
+        executorService.shutdownNow();
+    }
+
+    @Ignore
+    @Test
+    public void concurrentEventShouldNotSkipUidEmmitedDuringInitialization() throws Exception {
+        final AtomicInteger success = new AtomicInteger(0);
+        doAnswer(generateEmitEventAnswer(success))
+            .when(mailboxManager)
+            .addListener(eq(mailboxPath), any(MailboxListener.class), any(MailboxSession.class));
+
+        SelectedMailboxImpl selectedMailbox = new SelectedMailboxImpl(
+            mailboxManager,
+            imapSession,
+            mailboxPath);
+
+        assertThat(selectedMailbox.getLastUid().get()).isEqualTo(MESSAGE_UID_5);
+    }
+
+    @Ignore
+    @Test
+    public void concurrentEventShouldBeSupportedDuringInitialisation() throws Exception {
+        final AtomicInteger success = new AtomicInteger(0);
+        doAnswer(generateEmitEventAnswer(success))
+            .when(mailboxManager)
+            .addListener(eq(mailboxPath), any(MailboxListener.class), any(MailboxSession.class));
+
+        new SelectedMailboxImpl(
+            mailboxManager,
+            imapSession,
+            mailboxPath);
+
+        assertThat(success.get())
+            .as("Get the incremented value in case of successful event processing.")
+            .isEqualTo(1);
+    }
+
+    private Answer<Iterator<MessageUid>> sleepThenSearchAnswer() {
+        return new Answer<Iterator<MessageUid>>() {
+            @Override
+            public Iterator<MessageUid> answer(InvocationOnMock invocation) throws Throwable {
+                Thread.sleep(1000);
+                return ImmutableList.of(MessageUid.of(1), MessageUid.of(3)).iterator();
+            }
+        };
+    }
+
+    private Answer generateEmitEventAnswer(final AtomicInteger success) {
```

Comment:
I think `numOfSuccess` would be a better naming than `success` but if you always one call so one success maximu then keep success. But in this case consider changing it to boolean.